### PR TITLE
feat: Updating Python SDK to support Generator Steps & Asyncio handlers.

### PR DESCRIPTION
**Pull request by domderen**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -89,7 +89,7 @@ RUN mkdir /.cache /.local
 ADD runtimes/python3-9 /workspace
 RUN chown -R 9653 /.cache /.local /workspace
 WORKDIR /workspace
-RUN apk add git
+RUN apk add git gcc musl-dev
```
Comment (alexec) [inline Dockerfile:92]: these are some serious dependencies - and we're trying for minimal dependencies on the runtimes - what's the need for them?

Comment (domderen) [reply Dockerfile]: I modified this docker image to not use `git` dependency at all, by moving python dsl & sdk sources into that docker image and building them directly in the container. 

One of the changes I did in this PR, is modifying the SDK to support asyncio handlers, which also helped with implementing Generator Step handlers. Implementing it in asyncio fashion is I think cleaner than doing it with threads. But that meant that I need a dependency `aiohttp`. This dependency requires `gcc` & `musl-dev` to correctly build when installing it via pip. So in the Dockerfile, I specified that those two should be installed right before installing deps for the sdk, and then removed once sdk deps are built. I tested it and it seems to be working.

What do you think about such an approach to keeping the images small?

Comment (domderen) [reply Dockerfile]: This also speeds up `python3-9` runtime docker image startup time, as now it doesn't need to download dependencies on startup. Instead, those are already included in the image.

**Comment by domderen** [comment]
I'm not sure why test-examples is failing. It works just fine when I run it locally.

```bash
$ make test-examples
kubectl -n argo-dataflow-system apply -f config/apps/kafka.yaml
secret/dataflow-kafka-default configured
service/kafka-broker unchanged
service/zookeeper unchanged
statefulset.apps/kafka-broker unchanged
statefulset.apps/zookeeper unchanged
pod/input unchanged
kubectl -n argo-dataflow-system apply -f config/apps/stan.yaml
configmap/nats-config unchanged
configmap/stan-config unchanged
secret/dataflow-stan-default configured
service/nats unchanged
service/stan unchanged
statefulset.apps/nats configured
statefulset.apps/stan configured
kubectl -n argo-dataflow-system wait pod -l statefulset.kubernetes.io/pod-name --for condition=ready --timeout=2m
pod/testapi-0 condition met
pod/nats-0 condition met
pod/zookeeper-0 condition met
pod/kafka-broker-0 condition met
pod/stan-0 condition met
go test -timeout 20m -tags examples -v -count 1 ./examples
time="2021-09-08T02:54:40+02:00" level=info msg=cpu numCPU=16
time="2021-09-08T02:54:40+02:00" level=info msg=process pid=41152
time="2021-09-08T02:54:40+02:00" level=info msg=version version=v0.0.0-latest-0
=== RUN   Test
=== RUN   Test/101-hello
2021/09/08 02:54:40 deleting pipelines
2021/09/08 02:54:40 creating resource Pipeline/101-hello
2021/09/08 02:54:40 waiting for condition Running for 1m0s
2021/09/08 02:54:40 pipeline "101-hello" is  "" []
2021/09/08 02:54:40 pipeline "101-hello" is Pending "1 pending" []
2021/09/08 02:54:46 pipeline "101-hello" condition "Running" found
2021/09/08 02:54:46 waiting for condition SunkMessages for 1m30s
2021/09/08 02:54:46 pipeline "101-hello" is Running "1 running" ["Running"]
2021/09/08 02:54:55 pipeline "101-hello" condition "SunkMessages" found
=== RUN   Test/101-two-node
2021/09/08 02:54:55 deleting pipelines
2021/09/08 02:54:55 creating resource Pipeline/101-two-node
2021/09/08 02:54:55 waiting for condition Running for 1m0s
2021/09/08 02:54:55 pipeline "101-two-node" is  "" []
2021/09/08 02:54:55 pipeline "101-two-node" is Pending "1 pending" []
2021/09/08 02:54:55 pipeline "101-two-node" is Pending "2 pending" []
2021/09/08 02:54:56 pipeline "101-two-node" is Pending "1 pending, 1 running" []
2021/09/08 02:54:56 pipeline "101-two-node" condition "Running" found
2021/09/08 02:54:56 waiting for condition SunkMessages for 1m30s
2021/09/08 02:54:56 pipeline "101-two-node" is Running "2 running" ["Running"]
2021/09/08 02:55:04 pipeline "101-two-node" condition "SunkMessages" found
=== RUN   Test/102-filter
2021/09/08 02:55:04 deleting pipelines
2021/09/08 02:55:04 creating resource Pipeline/102-filter
2021/09/08 02:55:04 waiting for condition Running for 1m0s
2021/09/08 02:55:04 pipeline "102-filter" is  "" []
2021/09/08 02:55:04 pipeline "102-filter" is Pending "1 pending" []
2021/09/08 02:55:05 pipeline "102-filter" condition "Running" found
2021/09/08 02:55:05 waiting for condition SunkMessages for 1m30s
2021/09/08 02:55:05 pipeline "102-filter" is Running "1 running" ["Running"]
2021/09/08 02:55:11 pipeline "102-filter" condition "SunkMessages" found
=== RUN   Test/102-flatten-expand
2021/09/08 02:55:11 deleting pipelines
2021/09/08 02:55:11 creating resource Pipeline/102-flatten-expand
2021/09/08 02:55:11 waiting for condition Running for 1m0s
2021/09/08 02:55:11 pipeline "102-flatten-expand" is  "" []
2021/09/08 02:55:11 pipeline "102-flatten-expand" is Pending "2 pending" []
2021/09/08 02:55:11 pipeline "102-flatten-expand" is Pending "3 pending" []
2021/09/08 02:55:13 pipeline "102-flatten-expand" is Pending "1 pending, 2 running" []
2021/09/08 02:55:15 pipeline "102-flatten-expand" condition "Running" found
2021/09/08 02:55:15 waiting for condition SunkMessages for 1m30s
2021/09/08 02:55:15 pipeline "102-flatten-expand" is Running "3 running" ["Running"]
2021/09/08 02:55:19 pipeline "102-flatten-expand" condition "SunkMessages" found
=== RUN   Test/102-map
2021/09/08 02:55:19 deleting pipelines
2021/09/08 02:55:19 creating resource Pipeline/102-map
2021/09/08 02:55:19 waiting for condition Running for 1m0s
2021/09/08 02:55:19 pipeline "102-map" is  "" []
2021/09/08 02:55:19 pipeline "102-map" is Pending "1 pending" []
2021/09/08 02:55:23 pipeline "102-map" condition "Running" found
2021/09/08 02:55:23 waiting for condition SunkMessages for 1m30s
2021/09/08 02:55:23 pipeline "102-map" is Running "1 running" ["Running"]
2021/09/08 02:55:33 pipeline "102-map" condition "SunkMessages" found
=== RUN   Test/103-autoscaling
2021/09/08 02:55:33 deleting pipelines
2021/09/08 02:55:33 creating resource Pipeline/103-autoscaling
2021/09/08 02:55:33 waiting for condition Running for 1m0s
2021/09/08 02:55:33 pipeline "103-autoscaling" is  "" []
2021/09/08 02:55:33 pipeline "103-autoscaling" is Pending "1 pending" []
2021/09/08 02:55:35 pipeline "103-autoscaling" condition "Running" found
2021/09/08 02:55:35 waiting for condition SunkMessages for 1m30s
2021/09/08 02:55:35 pipeline "103-autoscaling" is Running "1 running" ["Running"]
2021/09/08 02:55:41 pipeline "103-autoscaling" condition "SunkMessages" found
=== RUN   Test/103-scaling
2021/09/08 02:55:41 deleting pipelines
2021/09/08 02:55:41 creating resource Pipeline/103-scaling
2021/09/08 02:55:41 waiting for condition Running for 1m0s
2021/09/08 02:55:41 pipeline "103-scaling" is  "" []
2021/09/08 02:55:41 pipeline "103-scaling" is Pending "1 pending" []
2021/09/08 02:55:44 pipeline "103-scaling" condition "Running" found
2021/09/08 02:55:44 waiting for condition SunkMessages for 1m30s
2021/09/08 02:55:44 pipeline "103-scaling" is Running "1 running" ["Running"]
2021/09/08 02:55:53 pipeline "103-scaling" condition "SunkMessages" found
=== RUN   Test/104-golang1-16
2021/09/08 02:55:53 deleting pipelines
2021/09/08 02:55:53 creating resource Pipeline/104-golang1-16
2021/09/08 02:55:53 waiting for condition Running for 1m0s
2021/09/08 02:55:53 pipeline "104-golang1-16" is  "" []
2021/09/08 02:55:53 pipeline "104-golang1-16" is Pending "1 pending" []
2021/09/08 02:56:06 pipeline "104-golang1-16" condition "Running" found
2021/09/08 02:56:06 waiting for condition SunkMessages for 1m30s
2021/09/08 02:56:06 pipeline "104-golang1-16" is Running "1 running" ["Running"]
2021/09/08 02:56:11 pipeline "104-golang1-16" condition "SunkMessages" found
=== RUN   Test/104-java16
2021/09/08 02:56:11 deleting pipelines
2021/09/08 02:56:11 creating resource Pipeline/104-java16
2021/09/08 02:56:11 waiting for condition Running for 1m0s
2021/09/08 02:56:11 pipeline "104-java16" is  "" []
2021/09/08 02:56:11 pipeline "104-java16" is Pending "1 pending" []
2021/09/08 02:56:26 pipeline "104-java16" condition "Running" found
2021/09/08 02:56:26 waiting for condition SunkMessages for 1m30s
2021/09/08 02:56:26 pipeline "104-java16" is Running "1 running" ["Running"]
2021/09/08 02:56:35 pipeline "104-java16" condition "SunkMessages" found
=== RUN   Test/104-node16
2021/09/08 02:56:35 deleting pipelines
2021/09/08 02:56:35 creating resource Pipeline/104-node16
2021/09/08 02:56:35 waiting for condition Running for 1m0s
2021/09/08 02:56:35 pipeline "104-node16" is  "" []
2021/09/08 02:56:35 pipeline "104-node16" is Pending "1 pending" []
2021/09/08 02:56:42 pipeline "104-node16" is Failed "1 failed" ["Completed"]
2021/09/08 02:57:02 pipeline "104-node16" condition "Running" found
2021/09/08 02:57:02 waiting for condition SunkMessages for 1m30s
2021/09/08 02:57:02 pipeline "104-node16" is Running "1 running" ["Running"]
2021/09/08 02:57:10 pipeline "104-node16" condition "SunkMessages" found
=== RUN   Test/104-python3-9
2021/09/08 02:57:10 deleting pipelines
2021/09/08 02:57:10 creating resource Pipeline/104-python3-9
2021/09/08 02:57:10 waiting for condition Running for 1m0s
2021/09/08 02:57:10 pipeline "104-python3-9" is  "" []
2021/09/08 02:57:10 pipeline "104-python3-9" is Pending "1 pending" []
2021/09/08 02:57:17 pipeline "104-python3-9" condition "Running" found
2021/09/08 02:57:17 waiting for condition SunkMessages for 1m30s
2021/09/08 02:57:17 pipeline "104-python3-9" is Running "1 running" ["Running"]
2021/09/08 02:57:25 pipeline "104-python3-9" condition "SunkMessages" found
=== RUN   Test/106-git-go
2021/09/08 02:57:25 deleting pipelines
2021/09/08 02:57:25 creating resource Pipeline/106-git-go
2021/09/08 02:57:25 waiting for condition Running for 1m0s
2021/09/08 02:57:25 pipeline "106-git-go" is  "" []
2021/09/08 02:57:25 pipeline "106-git-go" is Pending "1 pending" []
2021/09/08 02:57:29 pipeline "106-git-go" condition "Running" found
2021/09/08 02:57:29 waiting for condition SunkMessages for 1m30s
2021/09/08 02:57:29 pipeline "106-git-go" is Running "1 running" ["Running"]
2021/09/08 02:57:41 pipeline "106-git-go" condition "SunkMessages" found
=== RUN   Test/106-git-nodejs
2021/09/08 02:57:41 deleting pipelines
2021/09/08 02:57:41 creating resource Pipeline/106-git-nodejs
2021/09/08 02:57:41 waiting for condition Running for 1m0s
2021/09/08 02:57:41 pipeline "106-git-nodejs" is  "" []
2021/09/08 02:57:41 pipeline "106-git-nodejs" is Pending "1 pending" []
2021/09/08 02:57:45 pipeline "106-git-nodejs" condition "Running" found
2021/09/08 02:57:45 waiting for condition SunkMessages for 1m30s
2021/09/08 02:57:45 pipeline "106-git-nodejs" is Running "1 running" ["Running"]
2021/09/08 02:57:57 pipeline "106-git-nodejs" condition "SunkMessages" found
=== RUN   Test/106-git-python
2021/09/08 02:57:57 deleting pipelines
2021/09/08 02:57:57 creating resource Pipeline/106-git-python
2021/09/08 02:57:57 waiting for condition Running for 1m0s
2021/09/08 02:57:57 pipeline "106-git-python" is  "" []
2021/09/08 02:57:57 pipeline "106-git-python" is Pending "1 pending" []
2021/09/08 02:58:02 pipeline "106-git-python" condition "Running" found
2021/09/08 02:58:02 waiting for condition SunkMessages for 1m30s
2021/09/08 02:58:02 pipeline "106-git-python" is Running "1 running" ["Running"]
2021/09/08 02:58:19 pipeline "106-git-python" condition "SunkMessages" found
=== RUN   Test/107-completion
2021/09/08 02:58:19 deleting pipelines
2021/09/08 02:58:19 creating resource Pipeline/107-completion
2021/09/08 02:58:19 waiting for condition Running for 1m0s
2021/09/08 02:58:19 pipeline "107-completion" is  "" []
2021/09/08 02:58:19 pipeline "107-completion" is Pending "1 pending" []
2021/09/08 02:58:21 pipeline "107-completion" condition "Running" found
2021/09/08 02:58:21 waiting for condition Completed for 1m30s
2021/09/08 02:58:21 pipeline "107-completion" is Running "1 running" ["Running"]
2021/09/08 02:58:22 pipeline "107-completion" condition "Completed" found
=== RUN   Test/107-terminator
2021/09/08 02:58:22 deleting pipelines
2021/09/08 02:58:22 creating resource Pipeline/107-terminator
2021/09/08 02:58:22 waiting for condition Running for 1m0s
2021/09/08 02:58:22 pipeline "107-terminator" is  "" []
2021/09/08 02:58:22 pipeline "107-terminator" is Pending "1 pending" []
2021/09/08 02:58:22 pipeline "107-terminator" is Pending "2 pending" []
2021/09/08 02:58:24 pipeline "107-terminator" is Pending "1 running, 1 pending" []
2021/09/08 02:58:24 pipeline "107-terminator" is Pending "1 pending, 1 running" []
2021/09/08 02:58:24 pipeline "107-terminator" condition "Running" found
2021/09/08 02:58:24 waiting for condition Completed for 1m30s
2021/09/08 02:58:24 pipeline "107-terminator" is Running "2 running" ["Running"]
2021/09/08 02:58:25 pipeline "107-terminator" is Running "1 running, 1 succeeded, terminating" ["Running" "Terminating"]
2021/09/08 02:58:26 pipeline "107-terminator" condition "Completed" found
=== RUN   Test/108-container
2021/09/08 02:58:26 deleting pipelines
2021/09/08 02:58:26 creating resource Pipeline/108-container
2021/09/08 02:58:26 waiting for condition Running for 1m0s
2021/09/08 02:58:26 pipeline "108-container" is  "" []
2021/09/08 02:58:26 pipeline "108-container" is Pending "1 pending" []
2021/09/08 02:58:28 pipeline "108-container" condition "Running" found
2021/09/08 02:58:28 waiting for condition Completed for 1m30s
2021/09/08 02:58:28 pipeline "108-container" is Running "1 running" ["Running"]
2021/09/08 02:58:29 pipeline "108-container" condition "Completed" found
=== RUN   Test/108-fifos
2021/09/08 02:58:29 deleting pipelines
2021/09/08 02:58:29 creating resource Pipeline/108-fifos
2021/09/08 02:58:29 waiting for condition Running for 1m0s
2021/09/08 02:58:29 pipeline "108-fifos" is  "" []
2021/09/08 02:58:29 pipeline "108-fifos" is Pending "1 pending" []
2021/09/08 02:58:31 pipeline "108-fifos" condition "Running" found
2021/09/08 02:58:31 waiting for condition SunkMessages for 1m30s
2021/09/08 02:58:31 pipeline "108-fifos" is Running "1 running" ["Running"]
2021/09/08 02:58:36 pipeline "108-fifos" condition "SunkMessages" found
=== RUN   Test/109-group
2021/09/08 02:58:36 deleting pipelines
2021/09/08 02:58:36 creating resource Pipeline/109-group
2021/09/08 02:58:36 waiting for condition Running for 1m0s
2021/09/08 02:58:36 pipeline "109-group" is  "" []
2021/09/08 02:58:36 pipeline "109-group" is Pending "1 pending" []
2021/09/08 02:58:38 pipeline "109-group" condition "Running" found
2021/09/08 02:58:38 waiting for condition SunkMessages for 1m30s
2021/09/08 02:58:38 pipeline "109-group" is Running "1 running" ["Running"]
2021/09/08 02:58:46 pipeline "109-group" condition "SunkMessages" found
=== RUN   Test/301-cron-log
2021/09/08 02:58:46 deleting pipelines
2021/09/08 02:58:46 creating resource Pipeline/301-cron-log
2021/09/08 02:58:46 waiting for condition Running for 1m0s
2021/09/08 02:58:46 pipeline "301-cron-log" is  "" []
2021/09/08 02:58:46 pipeline "301-cron-log" is Pending "1 pending" []
2021/09/08 02:58:48 pipeline "301-cron-log" condition "Running" found
2021/09/08 02:58:48 waiting for condition SunkMessages for 1m30s
2021/09/08 02:58:48 pipeline "301-cron-log" is Running "1 running" ["Running"]
2021/09/08 02:58:58 pipeline "301-cron-log" condition "SunkMessages" found
=== RUN   Test/301-http
2021/09/08 02:58:58 deleting pipelines
2021/09/08 02:58:58 needs dataflow-103-http-main-source-default-secret.yaml
2021/09/08 02:58:58 creating resource Pipeline/301-http
2021/09/08 02:58:58 creating resource Secret/dataflow-http-main-source-default
2021/09/08 02:58:58 waiting for condition Running for 1m0s
2021/09/08 02:58:58 pipeline "301-http" is  "" []
2021/09/08 02:58:58 pipeline "301-http" is Pending "1 pending" []
2021/09/08 02:58:58 pipeline "301-http" is Pending "2 pending" []
2021/09/08 02:59:00 pipeline "301-http" is Pending "1 pending, 1 running" []
2021/09/08 02:59:01 pipeline "301-http" condition "Running" found
2021/09/08 02:59:01 waiting for condition SunkMessages for 1m30s
2021/09/08 02:59:01 pipeline "301-http" is Running "2 running" ["Running"]
2021/09/08 02:59:09 pipeline "301-http" condition "SunkMessages" found
=== RUN   Test/301-kafka
2021/09/08 02:59:09 deleting pipelines
2021/09/08 02:59:09 creating resource Pipeline/301-kafka
2021/09/08 02:59:09 waiting for condition Running for 1m0s
2021/09/08 02:59:09 pipeline "301-kafka" is  "" []
2021/09/08 02:59:09 pipeline "301-kafka" is Pending "1 pending" []
2021/09/08 02:59:11 pipeline "301-kafka" condition "Running" found
2021/09/08 02:59:11 waiting for condition SunkMessages for 1m30s
2021/09/08 02:59:11 pipeline "301-kafka" is Running "1 running" ["Running"]
2021/09/08 02:59:21 pipeline "301-kafka" condition "SunkMessages" found
=== RUN   Test/301-two-sinks
2021/09/08 02:59:21 deleting pipelines
2021/09/08 02:59:21 creating resource Pipeline/301-two-sinks
2021/09/08 02:59:21 waiting for condition Running for 1m0s
2021/09/08 02:59:21 pipeline "301-two-sinks" is  "" []
2021/09/08 02:59:21 pipeline "301-two-sinks" is Pending "1 pending" []
2021/09/08 02:59:23 pipeline "301-two-sinks" condition "Running" found
2021/09/08 02:59:23 waiting for condition SunkMessages for 1m30s
2021/09/08 02:59:23 pipeline "301-two-sinks" is Running "1 running" ["Running"]
2021/09/08 02:59:33 pipeline "301-two-sinks" condition "SunkMessages" found
=== RUN   Test/301-two-sources
2021/09/08 02:59:33 deleting pipelines
2021/09/08 02:59:33 creating resource Pipeline/301-two-sources
2021/09/08 02:59:33 waiting for condition Running for 1m0s
2021/09/08 02:59:33 pipeline "301-two-sources" is  "" []
2021/09/08 02:59:33 pipeline "301-two-sources" is Pending "1 pending" []
2021/09/08 02:59:35 pipeline "301-two-sources" condition "Running" found
2021/09/08 02:59:35 waiting for condition SunkMessages for 1m30s
2021/09/08 02:59:35 pipeline "301-two-sources" is Running "1 running" ["Running"]
2021/09/08 02:59:42 pipeline "301-two-sources" condition "SunkMessages" found
--- PASS: Test (302.39s)
    --- PASS: Test/101-hello (14.80s)
    --- PASS: Test/101-two-node (9.06s)
    --- PASS: Test/102-filter (7.65s)
    --- PASS: Test/102-flatten-expand (7.79s)
    --- PASS: Test/102-map (13.47s)
    --- PASS: Test/103-autoscaling (8.60s)
    --- PASS: Test/103-scaling (11.46s)
    --- PASS: Test/104-golang1-16 (18.70s)
    --- PASS: Test/104-java16 (23.95s)
    --- PASS: Test/104-node16 (34.59s)
    --- PASS: Test/104-python3-9 (14.91s)
    --- PASS: Test/106-git-go (16.14s)
    --- PASS: Test/106-git-nodejs (16.20s)
    --- PASS: Test/106-git-python (21.83s)
    --- PASS: Test/107-completion (3.08s)
    --- PASS: Test/107-terminator (4.08s)
    --- PASS: Test/108-container (2.99s)
    --- PASS: Test/108-fifos (6.66s)
    --- PASS: Test/109-group (10.07s)
    --- PASS: Test/301-cron-log (12.46s)
    --- PASS: Test/301-http (10.81s)
    --- PASS: Test/301-kafka (12.32s)
    --- PASS: Test/301-two-sinks (11.88s)
    --- PASS: Test/301-two-sources (8.88s)
PASS
```

**Comment by alexec** [comment]
```
    --- FAIL: Test/106-git-python-generator (150.03s)
    --- FAIL: Test/106-git-python (95.52s)
```

Can't merge with a newly failing job.

```
2021/09/08 12:06:43 creating resource Pipeline/106-git-python-generator
2021/09/08 12:06:43 waiting for condition Running for 1m0s
2021/09/08 12:06:43 pipeline "106-git-python-generator" is  "" []
2021/09/08 12:06:43 pipeline "106-git-python-generator" is Pending "1 pending" []
2021/09/08 12:07:02 pipeline "106-git-python-generator" is Failed "1 failed" ["Completed"]
2021/09/08 12:07:03 pipeline "106-git-python-generator" is Pending "1 pending" []
2021/09/08 12:07:17 pipeline "106-git-python-generator" is Failed "1 failed" ["Completed"]
2021/09/08 12:07:32 pipeline "106-git-python-generator" is Pending "1 pending" []
2021/09/08 12:07:43 timeout waiting for condition Running
    examples_test.go:104: 
        	Error Trace:	examples_test.go:104
        	            				examples_test.go:50
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	Test/106-git-python-generator
2021/09/08 12:07:43 waiting for condition SunkMessages for 1m30s
2021/09/08 12:07:43 pipeline "106-git-python-generator" is Pending "1 pending" []
2021/09/08 12:07:45 pipeline "106-git-python-generator" is Failed "1 failed" ["Completed"]
2021/09/08 12:08:23 pipeline "106-git-python-generator" is Pending "1 pending" []
2021/09/08 12:08:35 pipeline "106-git-python-generator" is Failed "1 failed" ["Completed"]
2021/09/08 12:09:13 timeout waiting for condition SunkMessages
    examples_test.go:104: 
        	Error Trace:	examples_test.go:104
        	            				examples_test.go:51
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	Test/106-git-python-generator
```

The pods are going into "Failed" state. Unfortunately, `test-examples` does not helpfully print the pod logs. I'll whip up a PR to fix that.

**Comment by domderen** [comment]
The reason it is failing, is because files from those git steps (./examples/git-python & ./examples/git-python-generator-step dirs) are not yet in argoproj-labs/argo-dataflow repository, so it is failing to initialize those two Steps. I tested it locally with those two Steps pointing at the right code in this commit: https://github.com/domderen/argo-dataflow/pull/3/commits/a6e3b89bd8ee8276881105b7b8a6e12d13001990

In that case those two steps worked correctly.

**Comment by alexec** [comment]
Lets suck it and see.

**Pull request merged**